src/sys/select.rs

```diff
@@ -58,15 +61,32 @@ impl FdSet {
     /// ```
     ///
     /// [`select`]: fn.select.html
-    pub fn highest(&mut self) -> Option<RawFd> {
-        for i in (0..FD_SETSIZE).rev() {
-            let i = i as RawFd;
-            if unsafe { libc::FD_ISSET(i, self as *mut _ as *mut libc::fd_set) } {
-                return Some(i)
-            }
-        }
+    pub fn highest(&self) -> Option<RawFd> {
+        self.fds().next_back()
+    }
 
-        None
+    /// Returns an iterator over the file descriptors in the set.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # extern crate nix;
+    /// # use nix::sys::select::FdSet;
+    /// # use std::os::unix::io::RawFd;
+    /// # fn main() {
```

Comment:
Usually you don't need the `fn main` line.  Do you really need it here?

Reply:
You're right. Removed.